crates/pop-contracts/src/utils/contracts_node.rs

```diff
@@ -47,28 +87,9 @@ pub async fn is_chain_alive(url: url::Url) -> Result<bool, Error> {
 /// * `output` - The optional log file for node output.
 ///
 pub async fn run_contracts_node(cache: PathBuf, output: Option<&File>) -> Result<Child, Error> {
-	let cached_file = cache.join(BIN_NAME);
-	if !cached_file.exists() {
-		let archive = archive_name_by_target()?;
-
-		let latest_version = latest_contract_node_release().await?;
-		let releases_url =
-			format!("{SUBSTRATE_CONTRACT_NODE}/releases/download/{latest_version}/{archive}");
-		// Download archive
-		let response = reqwest::get(releases_url.as_str()).await?.error_for_status()?;
-		let mut file = tempfile()?;
-		file.write_all(&response.bytes().await?)?;
-		file.seek(SeekFrom::Start(0))?;
-		// Extract contents
-		let tar = GzDecoder::new(file);
-		let mut archive = Archive::new(tar);
-		archive.unpack(cache.clone())?;
-		// Copy the file into the cache folder and remove the folder artifacts
-		let extracted_dir = cache.join(release_folder_by_target()?);
-		fs::copy(&extracted_dir.join(BIN_NAME), &cached_file)?;
-		fs::remove_dir_all(&extracted_dir.parent().unwrap_or(&cache.join("artifacts")))?;
-	}
-	let mut command = Command::new(cached_file.display().to_string().as_str());
+	let contracts_node = download_contracts_node(cache).await?;
```

Comment:
This seems weird - was this perhaps left in by mistake? Elsewhere we ask for permission to download, but here we just do it anyway.

Reply:
resolved. Please see latest commit and comment.